### PR TITLE
[GSoC] Added Pagination to Leaderboard

### LINKED ISSUE #3874: [GSoC Task] Add pagination in the leaderboard screen to load more users on scrolling down.
**Summary:** 
Add pagination in the leaderboard screen to load more users on scrolling down.

- [x] Implement Pagination

- [x] Add Button for AutoScroll to user’s rank

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by madhurgupta10**
**Description**
Fixes #3874

**Tests performed**
Tested betaDebug on Google Pixel 2 with API level 29.

### CODE REVIEW

**Comment by maskaravivek** [comment]
I get the following error while trying to build the project. I tried doing a clean build a couple of times. 

```
Duplicate class okhttp3.internal.ws.RealWebSocket found in modules jetified-okhttp-4.8.0.jar (com.squareup.okhttp3:okhttp:4.8.0) and jetified-okhttp-ws-3.4.1.jar (com.squareup.okhttp3:okhttp-ws:3.4.1)
Duplicate class okhttp3.internal.ws.WebSocketProtocol found in modules jetified-okhttp-4.8.0.jar (com.squareup.okhttp3:okhttp:4.8.0) and jetified-okhttp-ws-3.4.1.jar (com.squareup.okhttp3:okhttp-ws:3.4.1)
Duplicate class okhttp3.internal.ws.WebSocketReader found in modules jetified-okhttp-4.8.0.jar (com.squareup.okhttp3:okhttp:4.8.0) and jetified-okhttp-ws-3.4.1.jar (com.squareup.okhttp3:okhttp-ws:3.4.1)
Duplicate class okhttp3.internal.ws.WebSocketReader$FrameCallback found in modules jetified-okhttp-4.8.0.jar (com.squareup.okhttp3:okhttp:4.8.0) and jetified-okhttp-ws-3.4.1.jar (com.squareup.okhttp3:okhttp-ws:3.4.1)
Duplicate class okhttp3.internal.ws.WebSocketWriter found in modules jetified-okhttp-4.8.0.jar (com.squareup.okhttp3:okhttp:4.8.0) and jetified-okhttp-ws-3.4.1.jar (com.squareup.okhttp3:okhttp-ws:3.4.1)

Go to the documentation to learn how to Fix dependency resolution errors.

```

File: app/src/main/java/fr/free/nrw/commons/mwapi/OkHttpJsonApiClient.java [hunk 1]
```diff
@@ -65,32 +65,32 @@ public OkHttpJsonApiClient(OkHttpClient okHttpClient,
...
+        category,
+        limit,
+        offset);
+    HttpUrl.Builder urlBuilder = HttpUrl.parse(url).newBuilder();
+    urlBuilder.addQueryParameter("user", userName);
+    urlBuilder.addQueryParameter("duration", duration);
+    urlBuilder.addQueryParameter("category", category);
+    urlBuilder.addQueryParameter("limit", limit);
+    urlBuilder.addQueryParameter("offset", offset);
+    Timber.i("Url %s", urlBuilder.toString());
```
[the remark this answers is not in the document]
Comment (madhurgupta10) [reply app/src/main/java/fr/free/nrw/commons/mwapi/OkHttpJsonApiClient.java:84]: Okay

File: app/src/main/java/fr/free/nrw/commons/profile/leaderboard/DataSourceClass.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        return progressLiveStatus;
+    }
+
+    @Override
+    public void loadInitial(@NonNull LoadInitialParams<Integer> params,
+        @NonNull LoadInitialCallback<Integer, LeaderboardList> callback) {
+
+        compositeDisposable.add(okHttpJsonApiClient
+                .getLeaderboard(Objects.requireNonNull(sessionManager.getCurrentAccount()).name,
+                    "all_time", "upload", String.valueOf(PAGE_SIZE), String.valueOf(START_OFFSET))
```
[the remark this answers is not in the document]
Comment (madhurgupta10) [reply app/src/main/java/fr/free/nrw/commons/profile/leaderboard/DataSourceClass.java:41]: I plan to move these as constants when I implement the filters, will it be okay if we leave them as hardcoded for now?

**Comment by madhurgupta10** [comment]
> I get the following error while trying to build the project. I tried doing a clean build a couple of times.
> 
> ```
> Duplicate class okhttp3.internal.ws.RealWebSocket found in modules jetified-okhttp-4.8.0.jar (com.squareup.okhttp3:okhttp:4.8.0) and jetified-okhttp-ws-3.4.1.jar (com.squareup.okhttp3:okhttp-ws:3.4.1)
> Duplicate class okhttp3.internal.ws.WebSocketProtocol found in modules jetified-okhttp-4.8.0.jar (com.squareup.okhttp3:okhttp:4.8.0) and jetified-okhttp-ws-3.4.1.jar (com.squareup.okhttp3:okhttp-ws:3.4.1)
> Duplicate class okhttp3.internal.ws.WebSocketReader found in modules jetified-okhttp-4.8.0.jar (com.squareup.okhttp3:okhttp:4.8.0) and jetified-okhttp-ws-3.4.1.jar (com.squareup.okhttp3:okhttp-ws:3.4.1)
> Duplicate class okhttp3.internal.ws.WebSocketReader$FrameCallback found in modules jetified-okhttp-4.8.0.jar (com.squareup.okhttp3:okhttp:4.8.0) and jetified-okhttp-ws-3.4.1.jar (com.squareup.okhttp3:okhttp-ws:3.4.1)
> Duplicate class okhttp3.internal.ws.WebSocketWriter found in modules jetified-okhttp-4.8.0.jar (com.squareup.okhttp3:okhttp:4.8.0) and jetified-okhttp-ws-3.4.1.jar (com.squareup.okhttp3:okhttp-ws:3.4.1)
> 
> Go to the documentation to learn how to Fix dependency resolution errors.
> ```

Fixed

File: app/src/main/java/fr/free/nrw/commons/profile/leaderboard/DataSourceClass.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/profile/leaderboard/DataSourceClass.java:41]: It's OK for now if you don't forget to fix them soon :-)

**Pull request merged**